src/query/service/src/servers/http/v1/query/http_query.rs

```diff
@@ -139,25 +139,35 @@ pub struct HttpSessionConf {
 }
 
 impl HttpSessionConf {
+    // TODO(liyz): this function is not needed? Client side should apply the affect itself,
+    // and pass the db and settings parameters in the "session" field like:
+    // session: {database: "default", settings: {max_threads: "4"}}.
+    // do not actually need to apply these affect again on the server side.
     fn apply_affect(&self, affect: &QueryAffect) -> HttpSessionConf {
         let mut ret = self.clone();
         match affect {
             QueryAffect::UseDB { name } => {
                 ret.database = Some(name.to_string());
+                ret
             }
             QueryAffect::ChangeSettings {
                 keys,
                 values,
+                is_unset,
                 is_globals: _,
             } => {
+                // do not change the current settings if reset is true
+                if *is_unset {
+                    return ret;
```

Comment:
```suggestion
                    for key in keys.iter() {
                         settings.remove(key);
                    }
```

Reply:
I think this solve your problem

Reply:
I added a `settings.remove(key)` as you suggested, however it's not enough to resolve the issue. the client side still have to learn about when the settings was unsetted, or else the client will keep sending the settings stored in the client on the next time.

Reply:
>>>  client will keep sending the settings stored in the client on the next time.

this will not happen

```settings.remove(key); ```
this settings is sent to client,  client update local setting as a whole,

Reply:
the most simple example is in the client for sqllogict test

send https://github.com/datafuselabs/databend/blob/4784d7f31a3222b3e24b03059ec734521e6a4d61/tests/sqllogictests/src/client/http_client.rs#L128

recv
https://github.com/datafuselabs/databend/blob/4784d7f31a3222b3e24b03059ec734521e6a4d61/tests/sqllogictests/src/client/http_client.rs#L113

Reply:
oh, this makes sense. let me rethink about the design of this PR based this api contract.